docs/multiple-databases.md

```diff
@@ -0,0 +1,117 @@
+# Support for Multiple Databases
+
+For most use cases, a single Postgres database is sufficient to support a
+`graph-node` instance. When a `graph-node` instance outgrows a single
+Postgres database, it is possible to split the storage of `graph-node`'s
+data across multiple Postgres databases.
+
+Support for multiple databases is configured through a TOML configuration
+file. The location of the file is passed with the `--config` command line
+switch. When using a configuration file, it is not possible to use the
+options `--postgres-url`, `--postgres-secondary-hosts`, and
+`--postgres-host-weights`.
+
+The TOML file consists of two sections:
+* `[store]` describes the available databases
+* `[deployment]` describes how to place newly deployed subgraphs
+
+## Configuring Multiple Databases
+
+The `[store]` section must always have a primary store configured, which
+must be called `primary`. Each store can have additional read replicas that
+are used for responding to queries. Only queries are processed by read
+replicas. Indexing and block ingestion will always use the main database.
+
+Any number of additional stores, with their own read replicas, can also be
+configured. When read replicas are used, query traffic is split between the
+main database and the replicas according to their weights. In the example
+below, for the primary store, no queries will be sent to the main database,
+and the replicas will receive 50% of the traffic each. In the `vip` store,
+50% of the traffic go to the main database, and 50% to the replica.
+
+```toml
+[store]
+[store.primary]
+connection = "postgresql://graph:${PGPASSWORD}@primary/graph"
+weight = 0
+[store.primary.replicas.repl1]
+connection = "postgresql://graph:${PGPASSWORD}@primary-repl1/graph"
+weight = 1
+[store.primary.replicas.repl2]
+connection = "postgresql://graph:${PGPASSWORD}@primary-repl2/graph"
+weight = 1
```

Comment:
We could make the primary weight default to `0` and the others to `1`.

Reply:
I think that's just how we are doing it right now, but doubt it's how things should be set up in general; I wonder if we should even have a default. It seems important enough and easy enough to require setting it explicitly.